[hf seos] uninitialized variable?

Compilation error brought after 2d4129a5947d40f7071a3a95a1a1630538b6b13a in hf seos.

Bug raised on x86_64 and ARM64 devices.

***Compilation problems***
```bash
make -j all VERBOSE=1
```
Throws the following logs
```
===================================================================
Version info:      Iceman/master/v4.20728-263-g6f55a8a9c
Platform name:     Proxmark3 RDV4
PLATFORM:          PM3RDV4
PLATFORM_FPGA:     xc2s30
PLATFORM_SIZE:     512
Platform extras:   No extra selected
Included options:  SMARTCARD FLASH -DRDV4 LF HITAG EM4x50 EM4x70 ZX8211 GENERAL_HF ISO15693 LEGICRF ISO14443b ISO14443a ICLASS SEOS FELICA NFCBARCODE HFSNIFF HFPLOT COMPRESSION
Standalone mode:   LF_SAMYRUN
===================================================================
[*] MAKE client/all
[*] MAKE bootrom/all
[*] MAKE fpga_compress/all
[*] MAKE mfc_card_only/all
[*] MAKE mfc_card_reader/all
[*] MAKE mfd_aes_brute/all
[*] MAKE cryptorf/all
[-] CC fpga_compress.c
[-] CC ../../common/lz4/lz4hc.c
[-] CC ../../common/lz4/lz4.c
[-] CC mfkey32.c
[-] CC ../../../common/crapto1/crypto1.c
[-] CC nonce2key.c
[-] CC ../../../common/crapto1/crypto1.c
[-] CC cm.c
[-] CC ../../../common/crapto1/crapto1.c
[-] CC ../../../common/crapto1/crapto1.c
[-] CC ../../../common/bucketsort.c
[-] CC ../../common/cryptorf/cryptolib.c
[-] CC util.c
compiler version:  arm-none-eabi-gcc (15:12.2.rel1-1) 12.2.1 20221205
[=] CHECK version_pm3.c
[-] CC ../../../common/bucketsort.c
[-] CC nested_util.c
[-] CC staticnested_0nt.c
[-] CC iso14443crc.c
[-] CC sm.c
[-] CC ram-reset.s
[-] CC staticnested_1nt.c
[-] CC sleep.c
[-] CC util_posix.c
[-] CC mfkey32v2.c
[-] CC mfkey32nested.c
[-] CC mfkey64.c
[-] CXX sma.cpp
[-] CXX sma_multi.cpp
[-] CC mf_nonce_brute.c
[-] CC flash-reset.s
[-] CC staticnested_2nt.c
[-] CC ../common_arm/usb_cdc.c
[-] CC mf_trace_brute.c
[-] CC ../common_arm/clocks.c
[-] CC bootrom.c
[-] CC staticnested_2x1nt_rf08s_1key.c
[-] CC staticnested_2x1nt_rf08s.c
[-] CC ../common_arm/flashmem.c
[-] CC ../common_arm/ticks.c
===================================================================
[-] CC brute_key.c
[-] CC util_posix.c
[-] CC randoms.c
[-] CC version_pm3.c
[-] CC mfd_aes_brute.c
[-] CC mfd_multi_brute.c
Version info:      Iceman/master/v4.20728-263-g6f55a8a9c
Client platform:   Linux
GUI support:       QT5 found, enabled (Qt version 5.15.8 in /usr/lib/aarch64-linux-gnu)
native BT support: Bluez found, enabled
Jansson library:   system library not found, using local library
Lua library:       system library not found, using local library
Python3 library:   Python3 v3.11 found, enabled
GD library:        GD not found, disabled
Readline library:  enabled
Whereami library:  system library not found, using local library
Lua SWIG:          wrapper found
Python SWIG:       wrapper found
compiler version:  cc (Debian 12.2.0-14+deb12u1) 12.2.0
===================================================================
[-] CC src/mifare/aiddesfire.c
[-] CC src/aidsearch.c
[-] CC src/atrs.c
[-] CC src/cmdanalyse.c
[=] CXX brute_key
[-] CC src/cmdcrc.c
[-] CC src/cmddata.c
[-] CC src/cmdflashmem.c
[-] CC src/cmdflashmemspiffs.c
[-] CC src/cmdhf.c
[=] CXX mfd_aes_brute
[-] CC src/cmdhf14a.c
[-] CC src/pla.c
[-] CC src/cmdhf14b.c
[-] CC src/cmdhf15.c
[=] LD obj/bootrom.elf
[-] CC src/cmdhfcryptorf.c
[=] CXX mfd_multi_brute
[-] CC src/cmdhfepa.c
[=] GEN obj/bootrom.s19
[-] CC src/cmdhfemrtd.c
[-] CC src/cmdhffelica.c
[-] CC src/cmdhffido.c
[-] CC src/cmdhffudan.c
[-] CC src/cmdhfgallagher.c
[-] CC src/cmdhfksx6924.c
[-] CC src/cmdhfcipurse.c
[=] CXX cm
[=] CXX sm
[-] CC src/cmdhficlass.c
[-] CC src/cmdhfict.c
[-] CC src/cmdhflegic.c
[-] CC src/cmdhfjooki.c
[-] CC src/cmdhflist.c
[-] CC src/cmdhflto.c
[-] CC src/cmdhfmf.c
[-] CC src/cmdhfmfdes.c
[-] CC src/cmdhfmfhard.c
[-] CC src/cmdhfmfu.c
[-] CC src/cmdhfmfp.c
[-] CC src/cmdhfntag424.c
[-] CC src/cmdhfsaflok.c
[-] CC src/cmdhfseos.c
[-] CC src/cmdhfst.c
[-] CC src/cmdhfst25ta.c
[-] CC src/cmdhftesla.c
[-] CC src/cmdhfthinfilm.c
[-] CC src/cmdhftopaz.c
[-] CC src/cmdhftexkom.c
[-] CC src/cmdhfvas.c
[-] CC src/cmdhfxerox.c
[-] CC src/cmdhw.c
[-] CC src/cmdlf.c
[-] CC src/cmdlfawid.c
[-] CC src/cmdlfcotag.c
[-] CC src/cmdlfdestron.c
[-] CC src/cmdlfem.c
[-] CC src/cmdlfem410x.c
[-] CC src/cmdlfem4x05.c
[-] CC src/cmdlfem4x50.c
[-] CC src/cmdlfem4x70.c
[-] CC src/cmdlffdxb.c
[-] CC src/cmdlfguard.c
[-] CC src/cmdlfgallagher.c
[-] CC src/cmdlfhid.c
[-] CC src/cmdlfhitag.c
[-] CC src/cmdlfhitaghts.c
[-] CC src/cmdlfhitagu.c
[-] CC src/cmdlfidteck.c
[-] CC src/cmdlfindala.c
[-] CC src/cmdlfio.c
[-] CC src/cmdlfjablotron.c
[-] CC src/cmdlfkeri.c
[-] CC src/cmdlfmotorola.c
[-] CC src/cmdlfnedap.c
[-] CC src/cmdlfnexwatch.c
[-] CC src/cmdlfnoralsy.c
[-] CC src/cmdlfpac.c
[-] CC src/cmdlfparadox.c
[-] CC src/cmdlfpcf7931.c
[-] CC src/cmdlfpresco.c
[-] CC src/cmdlfpyramid.c
[-] CC src/cmdlfsecurakey.c
[-] CC src/cmdlft55xx.c
[-] CC src/cmdlfti.c
[-] CC src/cmdlfviking.c
[-] CC src/cmdlfvisa2000.c
[-] CC src/cmdlfzx8211.c
[-] CC src/cmdmain.c
src/cmdhfseos.c: In function ‘generate_command_wrapping.constprop’:
src/cmdhfseos.c:255:5: error: ‘*(long unsigned int *)(&mac[0])’ may be used uninitialized [-Werror=maybe-uninitialized]
  255 |     memcpy(completedCommand + command_header_len + ARRAYLEN(sizeofcommand) + ARRAYLEN(asn1_tag_cryptograph) + ARRAYLEN(padded_encrypted_Command) + ARRAYLEN(command_trailer) + ARRAYLEN(asn1_tag_mac), mac, ARRAYLEN(mac));
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/cmdhfseos.c:235:13: note: ‘*(long unsigned int *)(&mac[0])’ was declared here
  235 |     uint8_t mac[8];
      |             ^~~
[-] CC src/cmdmqtt.c
[-] CC src/cmdnfc.c
[-] CC src/cmdparser.c
[-] CC src/cmdpiv.c
[-] CC src/cmdscript.c
[-] CC src/cmdsmartcard.c
[-] CC src/cmdtrace.c
[-] CC src/cmdusart.c
[-] CC src/cmdwiegand.c
[-] CC src/comms.c
[-] CC src/crypto/asn1dump.c
[-] CC src/crypto/asn1utils.c
[-] CC src/crypto/libpcrypto.c
[-] CC src/crypto/originality.c
[-] CC src/emv/cmdemv.c
[-] CC src/emv/crypto.c
[-] CC src/emv/crypto_polarssl.c
[-] CC src/emv/dol.c
[-] CC src/emv/emv_pk.c
[-] CC src/emv/emv_pki.c
[-] CC src/emv/emv_pki_priv.c
[-] CC src/emv/emv_roca.c
[-] CC src/emv/emv_tags.c
[-] CC src/emv/emvcore.c
cc1: all warnings being treated as errors
[-] CC src/emv/emvjson.c
make[1]: *** [Makefile:1041: obj/cmdhfseos.o] Error 1
make[1]: *** Waiting for unfinished jobs....
[=] CXX sma
[=] CXX mfkey32
[=] CXX mfkey32v2
[=] CXX mfkey32nested
[=] CXX mfkey64
[=] CXX mf_nonce_brute
[=] CXX mf_trace_brute
[=] CXX sma_multi
[=] CXX nonce2key
[=] CXX staticnested_0nt
[=] CXX staticnested_1nt
[=] CXX staticnested_2nt
[=] CXX staticnested_2x1nt_rf08s_1key
[=] CXX staticnested_2x1nt_rf08s
[=] CXX fpga_compress
[*] MAKE armsrc/all
compiler version:  arm-none-eabi-gcc (15:12.2.rel1-1) 12.2.1 20221205
[-] GEN fpga_version_info.c
[-] GEN obj/fpga_all.bit.z
[-] CC start.c
[-] CC iso15693.c
[-] CC ../common/iso15693tools.c
[-] CC thinfilm.c
[-] CC lfops.c
[-] CC lfsampling.c
[-] CC pcf7931.c
[-] CC ../common/lfdemod.c
[-] CC lfadc.c
[-] CC ../common/lz4/lz4.c
[-] CC legicrf.c
[-] CC legicrfsim.c
make: *** [Makefile:174: client/all] Error 2
make: *** Waiting for unfinished jobs....
[-] CC ../common/legic_prng.c
[-] CC ../common_arm/flashmem.c
[-] CC i2c.c
[-] CC i2c_direct.c
[-] CC emvsim.c
[-] CC ../common/hitag2/hitag2_crypto.c
[-] CC hitag_common.c
[-] CC hitag2.c
[-] CC hitagS.c
[-] CC hitagu.c
[-] CC hitag2_crack.c
[-] CC em4x50.c
[-] CC ../common/bruteforce.c
[-] CC em4x70.c
[-] CC spiffs.c
[-] CC spiffs_cache.c
[-] CC spiffs_check.c
[-] CC spiffs_gc.c
[-] CC spiffs_nucleus.c
[-] CC spiffs_hydrogen.c
[-] CC hfops.c
[-] CC iso14443a.c
[-] CC mifareutil.c
[-] CC mifarecmd.c
[-] CC epa.c
[-] CC mifaresim.c
[-] CC sam_common.c
[-] CC sam_mfc.c
[-] CC sam_seos.c
[-] CC iso14443b.c
[-] CC ../common/crapto1/crypto1.c
[-] CC ../common/mbedtls/des.c
[-] CC desfire_crypto.c
[-] CC mifaredesfire.c
[-] CC ../common/mbedtls/aes.c
[-] CC ../common/mbedtls/platform_util.c
[-] CC iclass.c
[-] CC optimized_cipherutils.c
[-] CC optimized_ikeys.c
[-] CC optimized_elite.c
[-] CC optimized_cipher.c
[-] CC sam_picopass.c
[-] CC seos.c
[-] CC ../common/mbedtls/sha1.c
[-] CC ../common/mbedtls/sha256.c
[-] CC ../common/crc.c
[-] CC ../common/crc16.c
[-] CC ../common/crc32.c
[-] CC felica.c
[-] CC ../armsrc/Standalone/lf_samyrun.c
[-] CC lfzx.c
[-] CC appmain.c
[-] CC printf.c
[-] CC dbprint.c
[-] CC ../common/commonutil.c
[-] CC util.c
[-] CC string.c
[-] CC BigBuf.c
[-] CC ../common_arm/ticks.c
[-] CC ../common_arm/clocks.c
[-] CC hfsnoop.c
[-] CC ../common/generator.c
[-] CC cmac_calc.c
[-] CC cmac_3des.c
[-] CC fpgaloader.c
[-] CC ../common_arm/usb_cdc.c
[-] CC cmd.c
[-] CC fpga_version_info.c
[-] GEN obj/fpga_all.o
[-] CHECK version_pm3.c
[-] CC version_pm3.c
[=] LD obj/fullimage.stage1.elf
[-] GEN obj/fullimage.data.bin
[-] GEN obj/fullimage.data.bin.z
[=] LD obj/fullimage.elf
[=] GEN obj/fullimage.s19
```

--- a/client/src/cmdhfseos.c
+++ b/client/src/cmdhfseos.c
@@ -233,7 +233,11 @@ static void generate_command_wrapping(uint8_t *command_Header, int command_heade
     // 0181e43801010201 + 0000000000000001 + 0CCB3FFF800000000000000000000000 + 8510EB54DA90CB43AEE7FBFE816ECA25A10D + 9700 + 800000000000000000000000
 
     uint8_t mac[8];
-    create_cmac(diversified_mac_key, padded_toEncrypt, mac, padded_toEncrypt_len, sizeof(mac), encryption_algorithm);
+    memset(mac, 0x00, 8);
+    if (create_cmac(diversified_mac_key, padded_toEncrypt, mac, padded_toEncrypt_len, sizeof(mac), encryption_algorithm) != PM3_SUCCESS) {
+        PrintAndLogEx(ERR, _RED_("Failed to create MAC"));
+        return;
+    }
 
     // PrintAndLogEx(SUCCESS, "Encryption Key................... " _YELLOW_("%s"), sprint_hex_inrow(diversified_enc_key, 24));
     // PrintAndLogEx(SUCCESS, "MAC Key.......................... " _YELLOW_("%s"), sprint_hex_inrow(diversified_mac_key, 24));

